src/CppAst/CppTokenUtil.cs

```diff
@@ -23,21 +24,22 @@ public static void ParseCursorAttributs(CppGlobalDeclarationContainer globalCont
             if (tokenIt.CanPeek && tokenIt.PeekText() == "template")
                 SkipTemplates(tokenIt);
 
+            bool foundAttribute = false;
             while (tokenIt.CanPeek)
             {
                 if (ParseAttributes(globalContainer, tokenIt, ref attributes))
                 {
+                    foundAttribute = true;
                     continue;
                 }
 
-                // If we have a keyword, try to skip it and process following elements
-                // for example attribute put right after a struct __declspec(uuid("...")) Test {...}
-                if (tokenIt.Peek().Kind == CppTokenKind.Keyword)
+                if (foundAttribute && (tokenIt.Peek().Kind == CppTokenKind.Keyword || tokenIt.Peek().Kind == CppTokenKind.Identifier) ||
```

Comment:
This seems to be the crucial change of this PR, although the PR description helped a lot it is still difficult for me (and probably other developers) to parse this if-statement without context. Can you add some comment on this line given that the original comment was removed.

Reply:
I added a detailed comment about this.